datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -227,16 +227,25 @@ fn compute_array_dims(arr: Option<ArrayRef>) -> Result<Option<Vec<Option<u64>>>>
 }
 
 fn check_datatypes(name: &str, args: &[&ArrayRef]) -> Result<()> {
-    let data_type = args[0].data_type();
-    if !args.iter().all(|arg| {
-        arg.data_type().equals_datatype(data_type)
-            || arg.data_type().equals_datatype(&DataType::Null)
-    }) {
-        let types = args.iter().map(|arg| arg.data_type()).collect::<Vec<_>>();
-        return plan_err!("{name} received incompatible types: '{types:?}'.");
+    let data_types = args
+        .iter()
+        .map(|arg| arg.data_type())
+        .collect::<HashSet<_>>();
+    match data_types.len() {
+        1 => Ok(()),
+        2 => {
+            if data_types.contains(&DataType::Null) {
+                Ok(())
+            } else {
+                let types = args.iter().map(|arg| arg.data_type()).collect::<Vec<_>>();
```

Comment:
seems `types` variable is not used at all?

Reply:
Sorry, my bad, its it err message.
we may want to avoid code duplication for 
```
           let types = args.iter().map(|arg| arg.data_type()).collect::<Vec<_>>();
            plan_err!("{name} received incompatible types: '{types:?}'.")
```